build/Band_Representations.jl

```diff
@@ -0,0 +1,247 @@
+function site_symmetry_irreps_characters(sitegroup::SiteGroup{D}) where D
+    irreps=site_symmetry_irreps(sitegroup)
+    character_table=Matrix{Complex{Float64}}(undef, size(irreps[1].matrices)[1], size(irreps)[1] )
+    size(character_table)
+    for j in 1:size(irreps)[1]
+        for i in 1:size(irreps[1].matrices)[1]
+            character_table[i, j]=tr(irreps[j].matrices[i]) +0.00im +0.00
+        end
+    end
+character_table
+end
+
+
+"This function provides the induced band representation"
+function induced_band_representation(sitegroup::SiteGroup{D}, h::SymOperation{D}, ρ::Int64, k::KVec) where D
+    orbit_of_sitegroup=orbit(sitegroup, sitegroup.wp);
+    length_of_orbit=length(orbit_of_sitegroup)
+    #Now we sum over all the wyckoff positions in the orbit  
+    gα=cosets(sitegroup)
+    χ_G_k_G=0;
+    wp=sitegroup.wp;
+    wp_rvec=wp.qv.cnst
+    gα_inverse=gα[1]
+
+    transformed_k=transform(k, rotation(h)).cnst
+    for α′ in 1:length_of_orbit 
+
+        gα′=gα[α′]
+        tα′α′=rotation(h)*orbit_of_sitegroup[α′].qv.cnst+translation(h)-orbit_of_sitegroup[α′].qv.cnst
+        symop_t=SymOperation(one(Crystalline.SquareStaticMatrices.SqSMatrix{D,Float64}), -tα′α′ )
+
+        gα_inverse=inv(gα′)    
+
+        site_symmetry_index=in_site_symmetry(sitegroup, compose(gα_inverse, compose(symop_t, compose(h,  gα[α′], false), false ), false)   )
+      
+        if site_symmetry_index >0
+            χ_G_k_G=χ_G_k_G+exp(2im*π*dot(transformed_k, tα′α′))*tr(site_symmetry_irreps(sitegroup)[ρ].matrices[site_symmetry_index])
+        end
+    end
+    return χ_G_k_G
+end
+
+function subduce_onto_littlegroups(k::String, kvec::KVec, ρ::Int64, sitegroup::SiteGroup{D}) where D
+    sgnum=sitegroup.num
+    num_lgirreps= size(get_lgirreps(sgnum, D)[k])[1]
+    
+    size_lgirreps=size(get_lgirreps(sgnum, D)[k][1].matrices)[1]
+
+    character_table=Matrix{Complex{Float64}}(undef, size_lgirreps, num_lgirreps )
+
+    for j in 1:num_lgirreps
+        for i in 1:size_lgirreps
+            character_table[i, j]=tr(get_lgirreps(sgnum, D)[k][j].matrices[i])
+        end
+    end
+
+    XVector=Matrix{Complex{Float64}}(undef, (size_lgirreps, 1))
+
+    for i in 1:size_lgirreps
+        rep_index=group_representative(group(get_lgirreps(sgnum, D)[k][1])[i], spacegroup(sgnum, D)) 
+        iter=induced_band_representation(sitegroup, spacegroup(sgnum, D)[rep_index], ρ, kvec)
+        XVector[i, 1]= iter
+    end
+
+    return round.(character_table\XVector)
+
+end
+
+function return_irvec(sgnum::Int64, ρ::Int64, sitegroup::SiteGroup{D}, maximal::Bool) where D
+    maximal ? klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_maximal_lgs(sgnum, D)) : klab2kv = Dict(klab => kvec(lg) for (klab, lg) in get_littlegroups(sgnum, D))
+    irvec = Int[]
+    for (klab, kv) in klab2kv
+        kv_array   = subduce_onto_littlegroups(klab, kv, ρ, sitegroup)
+        append!(irvec, Int.(kv_array))
```

Comment:
Probably better to do the `Int.(...)` conversion in `subduce_onto_littlegroups` rather than outside it (this way, you are allocating two arrays instead of one, if you did `round.(Int, ...)` in `subduce_onto_littlegroups`).